app/actors/hyrax/actors/collections_membership_actor.rb

```diff
@@ -35,6 +36,12 @@ def assign_collections(env, collection_ids)
         def collections_without_edit_access(env)
           env.curation_concern.member_of_collections.select { |coll| env.current_ability.cannot?(:edit, coll) }
         end
+
+        # used to add participant permissions to work when there is only one collection at create time
```

Comment:
I'm not quite following why this is happening. I'm going to see if I can explain my understanding. Can you then amend the comment to explain in greater detail what is happening.

```ruby
# Given an array of collection_ids when it is size:
# * 0 set `env.attributes[:collection_id]` to `nil`
# * 1 do not set `env.attributes[:collection_id]`
# * 2 or greater `env.attributes[:collection_id]` is set to the first collection_id
#
# Later on, we will use `env.attributes[:collection_id] to apply the permissions of the associated collection to the created work.
```

Questions:

* What happens downstream when we have 0 or 1 collection ids?
* Why the first collection and not all of them? There seems to be an arbitrary decision to say the first collection is the permission granting collection. Could we have the union of those permissions applied later on?

Reply:
An adjustment on your interpretation...

```
# Given an array of collection_ids when it is size:
# * 0 do not set `env.attributes[:collection_id]`
# * 1 set `env.attributes[:collection_id]` to the one and only one collection
# * 2 do not set `env.attributes[:collection_id]`
#
# Later on in apply_permission_template_actor.rb, `env.attributes[:collection_id]` will be used to apply the permissions of the associated collection to the created work.
```

The original intention was that when a work is created directly in a collection, it will be assigned access based on the collection participants + access of its admin set.  There are several ways you can end up here on new...
* user clicked 'Add new work to this collection' button from the collection show page
* user started a new work from Dashboard -> Works and selected a collection on the relationships tab.  
  * If the user selects one collection on the relationships tab, the work is considered to be being created in that collection and access is added to the work based on the collection's participants.
  * If the user selects multiple collections on the relationships tab, the work is not considered to be being added directly to a collection and no collection participants are added to the work.

Because you can't tell whether you came from the collection show page or the dashboard -> works, the indicator of whether to add collection participants is that there is one and only one collection.

Reply:
RE: Arbitrary...  it is somewhat arbitrary to say we only add participants when there is one and only one collection.  I would have preferred to allow a user to specify a primary collection.  For the primary collection, participants will be added.  When a work belongs to multiple collections, the 'real' work is in the primary collection and all other references to the work would be aliases of the work.

But this is not backward compatible.  So it is not going in.  At least not in this sprint.